src/components/PaginationFooter/context/PaginationProvider.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { createContext, useContext, useMemo } from 'react';
+import type { stardust } from '@nebula.js/stardust';
+import type { FooterStyle, PaginationContentProps } from '../types';
+import getFooterStyle from '../../../utils/pagination-footer-utils';
+import useTranslations from '../../../hooks/useTranslations';
+
+interface IPaginationProvider extends PaginationContentProps {
+  footerStyle: FooterStyle;
+  tabIndex: number;
+  width: number;
+}
+
+interface PaginationProviderProps extends PaginationContentProps {
+  children: React.ReactNode;
+}
+
+export const areTabStopsEnabled = (keyboard: stardust.Keyboard) => !keyboard.enabled || keyboard.active;
+
+const NOOP = {} as IPaginationProvider;
+
+const PaginationProviderContext = createContext<IPaginationProvider>(NOOP);
+
+export const usePaginationContext = (): IPaginationProvider => useContext(PaginationProviderContext);
+
+const PaginationProvider = ({
+  children,
+  direction,
+  pageInfo,
+  handleChangeRowsPerPage,
+  footerContainer,
+  isSelectionMode,
+  handleChangePage,
+  totalRowCount,
+  totalColumnCount,
+  totalPages,
+  keyboard,
+  translator,
+  theme,
+  interactions,
+  rect,
+  layout,
+}: PaginationProviderProps): JSX.Element => {
+  // The elements can be focused in sequential keyboard navigation:
+  // - When nebula handles keyboard navigation
+  // and focus is somewhere inside the extension, or
+  // - When nebula does not handle keyboard navigation
+  const tabIndex = areTabStopsEnabled(keyboard) ? 0 : -1;
+  const footerStyle = useMemo(() => getFooterStyle(theme.background), [theme]);
+  const width = footerContainer ? footerContainer.getBoundingClientRect().width : rect.width;
+  const t = useTranslations({ translator });
+
+  const value = useMemo(
+    () => ({
+      direction,
+      pageInfo,
+      handleChangeRowsPerPage,
+      footerContainer,
+      isSelectionMode,
+      handleChangePage,
+      totalRowCount,
+      totalColumnCount,
+      totalPages,
+      keyboard,
+      translator: t,
+      theme,
+      interactions,
+      rect,
+      layout,
+      footerStyle,
+      tabIndex,
+      width,
+    }),
+    [
+      direction,
+      pageInfo,
+      handleChangeRowsPerPage,
+      footerContainer,
+      isSelectionMode,
+      handleChangePage,
+      totalRowCount,
+      totalColumnCount,
+      totalPages,
+      keyboard,
+      t,
+      theme,
+      interactions,
+      rect,
+      layout,
+      footerStyle,
+      tabIndex,
+      width,
+    ]
```

Comment:
when we have more dependencies like this one, i think useMemo kinda loses it's value because any change within that dependency list would cause a recalculation. 

we follow this pattern in Supernoval level also, which i'm personally not a big fan of tbh. I wish there was a better solution for this in general

Reply:
I agree, it's not ideal. But I cannot think of any way to actually solve it. Other then skipping the `useMemo`, since it probably not going to use the memoized value